### ISSUE TITLE
Signing with GPG produces non-compliant metadata

### ISSUE THREAD

**Issue by jku** [in-toto/in-toto#514]
Intoto seems to produce non-compliant metadata when signing with GPG. See https://github.com/in-toto/in-toto/blob/cb562165b2f62d1fabc80b0b5fdcf212df8fd3d6/tests/demo_files_gpg/write-code.8288ef56.link:
```
"signatures": [
  {
   "keyid": "8288ef560ed3795f9df2c0db56193089b285da58",
   "other_headers": "04000108001d1621048288ef560ed3795f9df2c0db56193089b285da5805025a25a335",
   "signature": "bc8db9988d722ccc2e34165be012488effec5f73be29d2db4f860154d90b58247ed8a589d13fc11b6301939e926353ecc0484ac14d55183e7c89ca90b315e9399103dcf7a41748f3ee4c55d114ac49fc32bc0f0a512e5324431eff2325555b24d4ae0426aba9cd0c5509ca14c1db2205da1d9ff7a50b1a5219f6fade5d4cd547552344191103d9511971b2e2cca7aa028fe5b3218c6ee3fa5beac94e67602fdd548e554b4bc42fedda42aa5e634b6805e4982ee0ae3c8b91515c61590e2fdce1e02b977e52913fd49e9c3e6c41f1a9eed80669fbb4a60d54cc70a77ada294e0bbe4c28c10312bb00754be626696abbf1b10c7d6ac952b591637f45a20ec26e8f"
  }
],
```
compare to the spec https://github.com/in-toto/docs/blob/master/in-toto-spec.md#42-file-formats-general-principles
```
  "signatures" : [
      { "keyid" : "<KEYID>",
        "sig" : "<SIGNATURE>" }, 
  "..."
  ]

```

Note `sig` vs `signature` as the keyname.

Signing with non-gpg keys produces compliant metadata ("sig" as keyname).

Origin: https://github.com/secure-systems-lab/securesystemslib/issues/448

**Comment by trishankatdatadog** [comment]
You are right: if you look at [production python-in-toto metadata](https://github.com/DataDog/integrations-core/tree/master/.in-toto), it says `"signature"`. But this is inconsistent with not only the in-toto spec, but also the sibling [TUF spec](https://theupdateframework.github.io/specification/v1.0.31/index.html#file-formats-general-principles), which both say `"sig"`.

I think we should update the spec to say so, but then this risks breaking backwards-compatibility for previous verifiers unless we update them, or new spec/verifiers accommodate both `"signature"` and `"sig"`. WDYT?

**Comment by jku** [comment]
I'm not sure what you mean by "updating the spec" but I think we agree on the backwards compat issues. 

If the issue is fixed (and "sig" is used consistently as the key):
* old verifiers (using current or older securesystemslib) will likely fail to verify new metadata. I don't see any way to work around this. It's unfortunate that the "other_headers" are required -- I think without those we might be able to make it work.
* new verifiers will fail to verify old metadata. This can probably be worked around: securesystemslib could keep accepting the old style signatures as well as the new ones

**Comment by trishankatdatadog** [comment]
> * old verifiers (using current or older securesystemslib) will likely fail to verify new metadata.

Oh, right, I forgot about this bit, unless we use a (soon to be deprecated) flag to allow generating both keys with the same value.

**Comment by lukpueh** [comment]
Thanks for reporting! Left a couple of related comments in https://github.com/secure-systems-lab/securesystemslib/issues/448#issuecomment-1300898410 pp. Let's bundle discussion in one place.
